scripts/gcloud.sh

```diff
@@ -165,117 +163,75 @@ gcloud_create_instance_template_if_not_exists() {
-  # support it
-  private_key="${7:-}"
-
-  # if set no additional arguments are used to start the container
-  no_args="${8:-}"
+  local announce="${6:-}"
+  # this parameter is mostly used on by CT nodes, although hoprd nodes also support it  
+  local private_key="${7:-}"
+  local mount_path="/app/hoprd-db"
+  local host_path="/var/hoprd"
+  local metadata_value=""
+  
+  log "checking for instance template ${name}"
+  if gcloud compute instance-templates describe "${name}" --quiet >/dev/null; then
+    log "instance template ${name} already present"
+    instance_group_name=$(echo $name | sed 's/\-[0-9].*//')    
+    if gcloud compute instance-groups describe ${gcloud_region} $instance_group_name --quiet >/dev/null; then
+      gcloud_delete_managed_instance_group $instance_group_name
+    fi
+    gcloud_delete_instance_template "${name}"
+  fi
 
-  skip_update_if_exists="${9:-false}"
+  metadata_value="google-logging-enabled=true"
+  metadata_value="${metadata_value},google-monitoring-enabled=true"
+  metadata_value="${metadata_value},enable-oslogin=true"
+  metadata_value="${metadata_value},startup-script='/opt/hoprd/startup-script.sh'"
+  metadata_value="${metadata_value},HOPRD_DOCKER_IMAGE=${image}"
 
-  args=""
-  # the environment is optional, since each docker image has a default environment set
-  if [ -n "${environment_id}" ]; then
-    args="--container-arg=\"--environment\" --container-arg=\"${environment_id}\""
+  if [ -n "${password}" ]; then
+    metadata_value="${metadata_value},HOPRD_PASSWORD=${password}"
   fi
 
   if [ -n "${api_token}" ]; then
-    extra_args="${extra_args} --container-arg=\"--apiToken\" --container-arg=\"${api_token}\""
+    metadata_value="${metadata_value},HOPRD_API_TOKEN=${api_token}"
   fi
 
-  if [ -n "${password}" ]; then
-    extra_args="${extra_args} --container-arg=\"--password\" --container-arg=\"${password}\""
+  # the environment is optional, since each docker image has a default environment set
+  if [ -n "${environment_id}" ]; then
+    metadata_value="${metadata_value},HOPRD_ENVIRONMENT_ID=${environment_id}"
   fi
 
+
   if [ -n "${private_key}" ]; then
-    extra_args="${extra_args} --container-arg=\"--privateKey\" --container-arg=\"${private_key}\""
+    metadata_value="${metadata_value},HOPRD_PRIVATE_KEY=\"--privateKey ${private_key}\""
   fi
 
   if [ -n "${announce}" ]; then
-    extra_args="${extra_args} --container-arg=\"--announce\""
+    metadata_value="${metadata_value},HOPRD_ANNOUNCE=--announce"
   fi
 
-  mount_path="/app/hoprd-db"
-  host_path="/var/hoprd"
-
-  log "checking for instance template ${name}"
-  if gcloud compute instance-templates describe "${name}" --quiet >/dev/null; then
-    log "instance template ${name} already present"
-
-    if [ "${skip_update_if_exists}" = "true" ]; then
-      # short-circuit, stop operation
-      return
-    fi
-
-    gcloud_delete_instance_template "${name}"
-  fi
+  echo ${metadata_value}
```

Comment:
Please use `log` if need to output something to the log.

Reply:
Solved